web/src/compose_setup.ts

```diff
@@ -57,6 +57,14 @@ function setup_compose_actions_hooks(): void {
     compose_actions.register_compose_cancel_hook(compose_call.abort_video_callbacks);
 }
 
+const throttled_update_draft = _.throttle(
+    () => {
+        drafts.update_draft({no_notify: true});
+    },
+    60000,
+    {leading: false, trailing: true},
+);
```

Comment:
I concern with this API is that you're not actually checking that the compose box hasn't closed and moved on to a different pursuit in the meantime. I think a clean method would look at the current `draft_id` for the compose box, compare that with what was passed into this function, and return early if that draft is already gone.

Doing that correctly may be a bit tricky, though, since the throttling function may not know how to specialize itself with the draft ID.

Reply:
Posted in [#issues > 📂 save drafts more? @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/.F0.9F.93.82.20save.20drafts.20more.3F/near/2305316). My current thinking is that probably this is the right code and we should just have a comment making the argument I posted in CZO.